cmake/Corrosion.cmake

```diff
@@ -1086,8 +1085,29 @@ function(corrosion_link_libraries target_name)
             $<TARGET_PROPERTY:${library},LINKER_LANGUAGE>
         )
 
-        corrosion_add_target_local_rustflags(${target_name} "-L$<TARGET_LINKER_FILE_DIR:${library}>")
-        corrosion_add_target_local_rustflags(${target_name} "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>")
+        if (TARGET ${library})
+            corrosion_add_target_local_rustflags(${target_name}
+                "-L$<TARGET_LINKER_FILE_DIR:${library}>"
+                "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>"
+            )
+            add_dependencies(_cargo-build_${target_name} ${library})
+            get_property(libs TARGET ${library} PROPERTY INTERFACE_LINK_LIBRARIES)
+            corrosion_link_libraries(${target_name} ${libs})
```

Comment:
Can you add a recursion check here? While ideally there would be no cyclic dependencies, I don't think that assumption holds in practice.

The implementation could probably use something like this (not tested)
 ```
set(recursion_list "${recursion_list};${library}")
if( "${library}" IN ${recursion_list})
    return
endif()
# do the things
# the pop happens implicitly, due to scoping of variable names in functions.
```